client/bgp_instance.go

```diff
@@ -23,7 +23,7 @@ func legacyBgpUnsupported(err error) bool {
 
 // BgpInstance Mikrotik resource
 type BgpInstance struct {
-	ID                       string `mikrotik:".id"`
+	Id                       string `mikrotik:".id"`
```

Comment:
@ddelnano 
This one is a breaking change to the `client` package.
I had to rename this field, as we are using `ID()` symbol for functions in the new `Resource` interface (it returns the value of `.id` field).

At this point, I think it was not the best design decision from my side to have 2 functions for each of the `ID/Find/Delete` fields, like `IDField()/ID()`, `FindField()/FindFieldValue()`, `DeleteField()/DeleteFieldValue()` as they are used **only** in pairs.
Now I think, it could be better to have a struct 
```go
type Field {
    Name  string
    Value string
}
```

and have simplified interfaces:
```go
type Finder interface {
    FindField() Field
}
```

or even move this meta info to resource tag like:
```go
type Bridge struct {
    ID   string `mikrotik:".id,id"` // id could be default to field with '.id' name
    Name string `mikrotik:"name,findID,deleteID"`
}
```

We can create a separate issue for discussion or use Discord if you think there are lot of open questions which require faster feedback than GitHub ping-pong

Reply:
Sorry for the late reply on this.

I'm much more concerned with making breaking changes to code exposed directly to the terraform end user. While it's possible someone is relying on it, I think the updates would be relatively minor. So I'm ok with introducing breaking `client` package changes (within reason) as long as you don't have concerns.

I see the value in the cleaner struct based approach you outlined above, but overall the status quo implementation is much better than our previous approach. So I don't have a strong opinion on that front especially since your outstanding PRs get us close (if not complete) to what is status quo now.

Reply:
I'm still refraining from merging this to continue the discussion, but it's approved if you want to proceed as is.

Reply:
I'll have another week with this thought in my head - maybe I cant find a better solution here (I can't think of any at the moment).